packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -107,11 +115,26 @@ export const plugin = <TContext>(
         requestContext:
           | GraphQLRequestContextExecutionDidStart<TContext>
           | GraphQLRequestContextDidEncounterErrors<TContext>,
+        reportTrace: boolean,
       ) {
         if (endDone) return;
         endDone = true;
         treeBuilder.stopTiming();
 
+        // Returning here if we aren't reporting the trace to make sure,
+        // endDone is set and the treeBuilder has stopped
+        // If we are running in gateway mode don't send the ftv1 trace
+        // after we have stopped trace timing.
+
+        // Return early if we aren't going to report the trace, but check
+        // `endDone` is true and the treebuilder has stopped.
+        // Also set captureTraces to false so if this is a gateway the
```

Comment:
I mean this is surely too late to actually accomplish anything (we're not going to be going downstream after didEnd, right?)

Reply:
This was mostly for `didResolveOperation`. This happens before execution so we can stop sending the header downstream if I understood the code correctly. I moved setting `captureTraces` to false into the `didResolveOperation` block

Reply:
I'm confused — this is the `didEnd` block, how is it before execution?